assets/js/editor-components/incompatible-extension-notice/index.tsx

```diff
@@ -36,15 +60,26 @@ export function IncompatibleExtensionsNotice( {
 		return null;
 	}
 
-	// console.log( incompatiblePaymentMethods );
+	const switchButtonLabel =
+		block === 'woocommerce/cart'
+			? __( 'Switch to classic cart', 'woo-gutenberg-products-block' )
+			: __(
+					'Switch to classic checkout',
+					'woo-gutenberg-products-block'
+			  );
+	const blockLabel = block === 'woocommerce/cart' ? 'cart' : 'checkout';
 
 	const noticeContent = (
 		<>
 			{ numberOfIncompatiblePaymentMethods > 1
 				? createInterpolateElement(
-						__(
-							'The following extensions may be incompatible with the block-based checkout. <a>Learn more</a>',
-							'woo-gutenberg-products-block'
+						sprintf(
+							// translators: %s is the name of the parent block.
+							__(
+								'Some extensions do not yet support the new %s and may impact the shopper experience. <a>Learn more</a>',
+								'woo-gutenberg-products-block'
+							),
+							blockLabel
```

Comment:
This is a nitpick but ideally you just do this string twice (above) without having cart/checkout as placeholder.

Cart is a single word in arabic, checkout is 2 words, depending on what comes before it, it might be translated with an article or not, but having it separately wouldn't make that obvious.

Reply:
https://github.com/woocommerce/woocommerce-blocks/pull/11228/commits/45d9543abe479064e7d9d87ad2180025852cc5a2